### PR TITLE
Use Prefs for feature flags

### PR DESCRIPTION
**Pull request by jasonLaster**
Associated Issue: #3819

### Summary of Changes

* Replaces devtools-config feature flags with Pref feature flags
* Removes the preview bottom bar (as it is old, difficult to implement, and likely needs UX)

### Next Steps

- Replace DevTools Launchpad Feature flagging with a pref based approach

### CODE REVIEW

File: docs/feature-flags.md [hunk 1]
```diff
@@ -29,11 +29,11 @@ Search for [isEnabled](https://github.com/devtools-html/debugger.html/search?utf
 
 ```js
 // within the components directory you can require isEnabled
-const { isEnabled } = require("devtools-config");
+const { features } = require("devtools-config");
 
 // feature check can be done in render() method
 render() {
-  if (!isEnabled("pokemon-go")) {
+  if (!features.pokemon-go) {
```
Comment (wldcordeiro) [inline docs/feature-flags.md]: `if (!features.pokemon-go)` doesn't make sense as valid JS.

File: src/utils/prefs.js [hunk 2]
```diff
@@ -70,18 +75,25 @@ export const prefs = new PrefsHelper("devtools", {
...
+  mapScopes: ["Bool", "map-scopes"],
+  breakpointsDropdown: ["Bool", "breakpoints-dropdown"],
+  removeCommandBarOptions: ["Bool", "remove-command-bar-options"],
+  workers: ["Bool", "workers"],
+  codeCoverage: ["Bool", "code-coverage"],
+  eventListeners: ["Bool", "event-listeners"],
+  outline: ["Bool", "outline"]
 });
 
+// export function isEnabled(key: string) {
```
Comment (wldcordeiro) [inline src/utils/prefs.js]: Commented out?

File: src/utils/breakpoint/index.js [hunk 3]
```diff
@@ -94,7 +94,7 @@ export function breakpointAtLocation(
 
     // NOTE: when column breakpoints are disabled we want to find
     // the first breakpoint
-    if (!features.columnBreakpoints) {
+    if (!features.olumnBreakpoints) {
```
Comment (bomsy) [inline src/utils/breakpoint/index.js]: typo here.

**Comment by wldcordeiro** [comment]
This is passing now after I rebuilt with no cache on CircleCI @jasonLaster

**Comment by jasonLaster** [comment]
https://treeherder.mozilla.org/#/jobs?repo=try&revision=109acb1bff9348aa2f00ff52c055a56b1c9069da

**Pull request merged**